server/src/main/java/org/apache/druid/client/coordinator/CoordinatorClient.java

```diff
@@ -58,6 +59,12 @@ public interface CoordinatorClient
    */
   ListenableFuture<List<DataSourceInformation>> fetchDataSourceInformation(Set<String> datasources);
 
+  /**
+   * Fetch bootstrap segments from the coordiantor. The results must be streamed back to the caller as the
+   * result set can be large.
+   */
+  ListenableFuture<CloseableIterator<DataSegment>> fetchBootstrapSegments();
```

Comment:
If we want this API to evolve in the future to be able to fully bootstrap historicals, I think only streaming back the set of segments might not suffice. We would also need some kind of ID or hostname to be sent to the clone historical. So, maybe we should have a `BootstrapSegmentResponse` class.

__Explanation__
The typical way this would work is a historical would try to clone another historical. For that, the coordinator would need to keep track of which historicals need to be cloned, which are currently being cloned. Once cloned, the historical would need to report back that it has cloned a certain historical. All of this would require some ID or hostname of the original historical to be sent back to the clone when it requests the bootstrap segments.

__Alternatively (and I would prefer this), let's just focus the current API on bootstrap broadcast segments only. When we have a full fledged design for bootstrapping all segments of a historical, we can add a new API for it.
In that spirit, I think we should rename the API to `/bootstrapBroadcastSegments`.

Reply:
Hmm, I was thinking that a new historical would actually pass the server info, tier, etc. in the payload when it talks to the coordinator.

Agreed that once we have a full design and idea on the upgrade situation, we can perhaps add a new API then. I can change the API semantics a bit tailored to the current  situation.

Reply:
Actually, I kept the same API name but added a `BootstrapSegmentsInfo` container similar to what you were suggesting. This should make it extensible, but at worst, we can still evolve the API, keeping the name with a different version number or so.

Adding more specific endpoints like `/bootstrapBroadcastSegments` would make it hard to keep track of and need to be deprecated eventually. Let me know if that seems like a reasonable middle ground.

Reply:
Yeah, that works for me.